File: 백준/Silver/1063. 킹/킹.py
import sys
input = sys.stdin.readline

move = {
    "T": [1,0],
    "B": [-1,0],
    "L": [0,-1],
    "R": [0,1],
    "LT": [1,-1],
    "RT": [1,1],
    "LB": [-1,-1],
    "RB": [-1,1],
    }

k,s,n = input().rstrip('\n').split()
k_pos,s_pos = [int(k[1]),ord(k[0])-64], [int(s[1]),ord(s[0])-64]

for _ in range(int(n)):
    y,x = move[input().rstrip('\n')]
    if 1<=k_pos[0]+y<=8 and 1<=k_pos[1]+x<=8:
        if k_pos[0]+y == s_pos[0] and k_pos[1]+x == s_pos[1] and 1<=s_pos[0]+y<=8 and 1<=s_pos[1]+x<=8:
            s_pos = [s_pos[0]+y,s_pos[1]+x]
        elif k_pos[0]+y == s_pos[0] and k_pos[1]+x == s_pos[1] and not(1<=s_pos[0]+y<=8 and 1<=s_pos[1]+x<=8): continue
        k_pos = [k_pos[0]+y,k_pos[1]+x]
print(chr(k_pos[1]+64)+str(k_pos[0]))
print(chr(s_pos[1]+64)+str(s_pos[0]))

File: 백준/Silver/1063. 킹/킹.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        HashMap<String, Integer[]> move = new HashMap<>();
        move.put("T", new Integer[]{1, 0});
        move.put("B", new Integer[]{-1, 0});
        move.put("L", new Integer[]{0, -1});
        move.put("R", new Integer[]{0, 1});
        move.put("LT", new Integer[]{1, -1});
        move.put("RT", new Integer[]{1, 1});
        move.put("LB", new Integer[]{-1, -1});
        move.put("RB", new Integer[]{-1, 1});

        StringTokenizer st1 = new StringTokenizer(br.readLine());
        String k = st1.nextToken();
        String s = st1.nextToken();
        String n = st1.nextToken();

        ArrayList<Integer> k_pos = new ArrayList<>(Arrays.asList(Integer.parseInt(k.substring(1)), (int) k.charAt(0) - 64));
        ArrayList<Integer> s_pos = new ArrayList<>(Arrays.asList(Integer.parseInt(s.substring(1)), (int) s.charAt(0) - 64));

        for (int i = 0; i < Integer.parseInt(n); i++) {
            StringTokenizer st2 = new StringTokenizer(br.readLine());
            Integer[] next = move.get(st2.nextToken());
            if ((1 <= k_pos.get(0) + next[0] && k_pos.get(0) + next[0] <= 8) && 1 <= k_pos.get(1) + next[1] && k_pos.get(1) + next[1] <= 8) {
                if ((k_pos.get(0) + next[0] == s_pos.get(0) && k_pos.get(1) + next[1] == s_pos.get(1)) && ((1 <= s_pos.get(0) + next[0] && s_pos.get(0) + next[0] <= 8) && 1 <= s_pos.get(1) + next[1] && s_pos.get(1) + next[1] <= 8)) {
                    s_pos.set(0, s_pos.get(0) + next[0]);
                    s_pos.set(1, s_pos.get(1) + next[1]);
                } else if ((k_pos.get(0) + next[0] == s_pos.get(0) && k_pos.get(1) + next[1] == s_pos.get(1)) && !((1 <= s_pos.get(0) + next[0] && s_pos.get(0) + next[0] <= 8) && 1 <= s_pos.get(1) + next[1] && s_pos.get(1) + next[1] <= 8)) {
                    continue;
                }
                k_pos.set(0, k_pos.get(0) + next[0]);
                k_pos.set(1, k_pos.get(1) + next[1]);
            }
        }

        System.out.print((char) (k_pos.get(1) + 64));
        System.out.println(k_pos.get(0));
        System.out.print((char) (s_pos.get(1) + 64));
        System.out.println(s_pos.get(0));
    }
}
